sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -198,6 +203,66 @@ case class StringSplit(str: Expression, pattern: Expression)
   override def prettyName: String = "split"
 }
 
+/**
+ * Splits a string into arrays of sentences, where each sentence is an array of words.
+ * The 'lang' and 'country' arguments are optional, and if omitted, the default locale is used.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(str, lang, country) - Splits str into an array of array of words.",
+  extended = "> SELECT _FUNC_('Hi there! Good morning.');\n  [['Hi','there'], ['Good','morning']]")
+case class Sentences(
+    str: Expression,
+    language: Expression = Literal(""),
+    country: Expression = Literal(""))
+  extends Expression with ImplicitCastInputTypes with CodegenFallback {
+
+  def this(str: Expression) = this(str, Literal(""), Literal(""))
+  def this(str: Expression, language: Expression) = this(str, language, Literal(""))
+
+  override def nullable: Boolean = true
+  override def dataType: DataType =
+    ArrayType(ArrayType(StringType, containsNull = false), containsNull = false)
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, StringType, StringType)
+  override def children: Seq[Expression] = str :: language :: country :: Nil
+
+  override def eval(input: InternalRow): Any = {
+    val string = str.eval(input)
+    if (string == null) {
+      null
+    } else {
+      val locale = try {
+        new Locale(language.eval(input).asInstanceOf[UTF8String].toString,
+          country.eval(input).asInstanceOf[UTF8String].toString)
+      } catch {
+        case _: NullPointerException | _: ClassCastException => Locale.getDefault
```

Comment:
Oh, it was ambiguous. `new Locale("abc", "xyz")` is created, it's not null. The following works but uses JVM Default locale. `BreakIterator.getSentenceInstance` does not return `null` with wrong locale instance.
```
scala> val x = java.text.BreakIterator.getSentenceInstance(new java.util.Locale("xxx", "bbb"))
x: java.text.BreakIterator = [checksum=0xcba403eb]
```